lib/webdrivers/mswebdriver.rb

```diff
@@ -1,38 +1,84 @@
     class << self
+      attr_accessor :ignore, :file_name
+
+      def current_version
+        raise NotImplementedError, 'Unable to programatically determine the version of most MicrosoftWebDriver.exe'
+      end
+
       def windows_version
-        Webdrivers.logger.debug 'Checking current version'
+        return @windows_version if @windows_version
 
         # current_version() from other webdrivers returns the version from the webdriver EXE.
         # Unfortunately, MicrosoftWebDriver.exe does not have an option to get the version.
         # To work around it we query the currently installed version of Microsoft Edge instead
         # and compare it to the list of available downloads.
-        version = `powershell (Get-AppxPackage -Name Microsoft.MicrosoftEdge).Version`
-        raise 'Failed to check Microsoft Edge version.' if version.empty? # Package name changed?
+        version = system_call('powershell (Get-AppxPackage -Name Microsoft.MicrosoftEdge).Version')
+        raise VersionError, 'Failed to check Microsoft Edge version' if version.empty? # Package name changed?
+
+        Webdrivers.logger.debug "Current version of Microsoft Edge is #{version}"
+
+        @windows_version = Gem::Version.new(version)
+      end
+
+      def latest_version
+        return @latest_version if @latest_version
+
+        if windows_version.segments.first == 45
+          raise VersionError, 'Webdrivers is unable to provide this driver; Run this command: '\
+          '`DISM.exe /Online /Add-Capability /CapabilityName:Microsoft.WebDriver~~~~0.0.1.0` '\
+          'as discussed here: https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/#downloads; '\
+          'Also please set `Webdrivers::MSWebdriver.ignore = true`'
+        end
 
-        Webdrivers.logger.debug "Current version of Microsoft Edge is #{version.chomp!}"
+        version = if windows_version < Gem::Version.new(44)
+                    @file_name = 'MicrosoftWebDriver.exe'
+                    windows_version.segments[1]
+                  else
+                    @file_name = 'msedgedriver.exe'
+                    '75'
+                  end
 
-        build = version.split('.')[1] # "41.16299.248.0" => "16299"
-        Webdrivers.logger.debug "Expecting MicrosoftWebDriver.exe version #{build}"
-        build.to_i
+        Webdrivers.logger.debug "Desired build of Microsoft WebDriver is #{version}"
+
+        @latest_version = Gem::Version.new(version)
       end
 
-      # Webdriver binaries for Microsoft Edge are not backwards compatible.
-      # For this reason, instead of downloading the latest binary we download the correct one for the
-      # currently installed browser version.
-      alias version windows_version
+      def remove
+        @download_url = nil
+        @latest_version = nil
+        Webdrivers.logger.debug 'Deleting MicrosoftWebDriver.exe'
+        FileUtils.rm_f(File.join(install_dir, 'MicrosoftWebDriver.exe'))
```

Comment:
The _f version doesn't return errors if file can't be deleted - Not sure if that's a concern or not

Reply:
Yeah, same as #87.

Reply:
Removed this code